.github/templates/main.yml

```diff
@@ -72,6 +73,34 @@ steps:
       retention-days: 1
 #@ end
 
+#@ def buildNugetPkgs():
```

Comment:
Let's split this into three functions - `buildPackages()` returns the nodes for actually building the packages, `findPackageVersion()` returns the node for looking up the package version, and finally, `uploadPackageArtifacts()` will return the nodes for uploading the artifacts.

Reply:
I would not do `findPackageVersion` a function since it isn't not reused and has "no loop" to perform.

Reply:
It's more about reducing the complexity of the parent function than code reuse -  since the body of `find-nupkg-version` is a bit of a mouthful and not really interesting, I'd just like to avoid reading it 😅